src/test/scala/exercise3/TestSynthesizerZio.scala

```diff
@@ -0,0 +1,40 @@
+package exercise3
+
+import exercise3.SynthesizerZIO.whiteNoise
+import zio.test.Assertion.{approximatelyEquals, equalTo}
+import zio.test.environment.TestRandom
+import zio.test.{DefaultRunnableSpec, assert}
+
+object TestSynthesizerZio extends DefaultRunnableSpec {
+  def spec = suite("WhiteNoise Test")(
+    testM("WhiteNoise without parameters should return array of Random Double Values between -0.5 and 0.5") {
+      for {
+        queue <- whiteNoise()
+      } yield assert(queue.size)(equalTo(440)) &&
+        assert(queue.head)(approximatelyEquals(0, 0.5)) &&
+        assert(queue.last)(approximatelyEquals(0, 0.5))
+    },
+    testM("WhiteNoise with frequency parameter should return queue of different length") {
+      val newFrequency: Double = 220.0
+      val newFrequencyQueueLength: Int = (440.0 * (440.0 / newFrequency)).toInt
+      for {
+        queue <- whiteNoise(frequency = newFrequency)
+      } yield assert(queue.size)(equalTo(newFrequencyQueueLength)) &&
+        assert(queue.head)(approximatelyEquals(0, 0.5))
+    },
+    testM("WhiteNoise with different volume parameter should return queue with appropriately different values") {
+      for {
+        queue <- whiteNoise(volume = 40)
+      } yield assert(queue.head)(approximatelyEquals(0, 20.0)) &&
+        assert(queue.last)(approximatelyEquals(0, 20.0))
+    },
+    testM("WhiteNoise with random feed should return the preset elements") {
+      for {
+        _ <- TestRandom.feedDoubles(0.4, -0.4, 0.2, 0.3)
+        queue <- whiteNoise()
+      } yield assert(queue.head)(equalTo(0.4)) &&
```

Comment:
Since both the queue and stacks are case classes could just do
```scala
val expectedQueue = DoubleStackQueue[Double]().enqueue(0.4).enqueue(-0.4).enqueue(0.2).enqueue(0.3)
...
assert(queue)(equalTo(expectedQueue))
```
, which is also not ideal, but a bit less abominable.

If we had a companion object, as all standard collections do,
```scala
object Queue{
  def apply[T](l:List[T]): Queue[T] = ???
```
we could just do `val expectedQueue  = Queue(0.4, -0.4, 0.2, 0.3)`